src/yunohost/tools.py

```diff
@@ -565,3 +565,25 @@ def tools_diagnosis(auth, private=False):
         diagnosis['private']['domains'] = domain_list(auth)['domains']
 
     return diagnosis
+
+def tools_public_key():
+    key_path = '/root/.ssh/id_rsa_ynh'
+    pub = os.path.exists(key_path+'.pub')
+    private = os.path.exists(key_path)
+
+    if pub and (not private):
+        os.remove(key_path+'.pub')
+        pub = not pub
+    elif (not pub) and private:
+        os.remove(key_path)
+        private = not private
+
+    if (not pub) and (not private):
+        os.system('ssh-keygen -P "" -f %s >/dev/null' % key_path)
```

Comment:
- Use strong default to `ssh-keygen` command, like `-t rsa -b 4096` or the recent `-t ed25519`
- Add a comment, like `-c "Generated by YunoHost"`.
- Be quiet : `-q`

Some worth reading: https://wiki.mozilla.org/Security/Guidelines/OpenSSH#Key_generation & https://wiki.mozilla.org/Security/Key_Management